### PR TITLE
Update documentation

### PR DESCRIPTION
**Pull request by atodorov**
Several improvements to documentation.

@tkdchen I have added sections on how to configure Nitrate with Gunicorn and serve it as a Docker image in Google Cloud Engine. I have one such setup which I created using the documented instructions and I wanted to share them. 

I also have another instance of Nitrate hosted on Red Hat's OpenShift cloud and I will create another file documenting how to get that up and running.

### CODE REVIEW

File: docs/source/set_dev_env.rst [hunk 1]
```diff
@@ -41,19 +41,26 @@ Create database and user::
...
 set in `tcms/settings/devel.py` with default value. Each of them is `nirrate`.
 
-Load database schema::
+.. note::
 
-    mysql -uroot nitrate < contrib/sql/nitrate_db_setup.sql
+    You may want to adjust the database and/or other configuration settings.
+    Override them in ``./tcms/settings/devel.py`` if necessary. While the
+    Nitrate team prefers MySQL, sqlite has been verified to work for development
+    and some people have reported using PostgreSQL in production!
```
Comment (tkdchen) [inline docs/source/set_dev_env.rst]: I don't know anyone run Nitrate with PostgreSQL. Actually, there are some code, which is raw SQL statements, only specific to MySQL in ``report`` app.

Comment (atodorov) [inline docs/source/set_dev_env.rst]: I have one instance of Nitrate with PostgreSQL and it is running fine regardless of the hard-coded SQL statements. I agree that the SQl makes it less portable though. Maybe a sentence highlighting this fact needs to be added, what do you think ?

Comment (tkdchen) [inline docs/source/set_dev_env.rst]: > Maybe a sentence highlighting this fact needs to be added, what do you think ?

👍  Good idea.

File: docs/source/set_dev_env.rst [hunk 2]
```diff
@@ -41,19 +41,29 @@ Create database and user::
...
 
-Load database schema::
+.. note::
 
-    mysql -uroot nitrate < contrib/sql/nitrate_db_setup.sql
+    You may want to adjust the database and/or other configuration settings.
+    Override them in ``./tcms/settings/devel.py`` if necessary. While the
+    Nitrate team prefers MySQL, sqlite appears to work for development
+    and some people have used PostgreSQL with varying success in production!
+    At the the moment Nitrate is not 100% portable between database backends
```
Comment (tkdchen) [inline docs/source/set_dev_env.rst]: Should be "At the moment", right?

**Pull request merged**